### DISCUSSION TITLE
Node 16 "type" = module

### DISCUSSION THREAD

**Discussion by kevinswarner** [dynamoose/dynamoose#1524] [Q&A]
I have a simple Node 16 project set to type="module". I am not using Babel or any compiler. When attempting to use es import and use Dynamoose in this environment, things just don't work. When I switch to type="commonjs" and use require, things work. For example, this does not work...

```
import * as dynamoose from 'dynamoose';

const schema = new dynamoose.Schema(...);
```

I get the error...

TypeError: dynamoose.Schema is not a constructor

I am still not familiar with the complexities of commonjs vs module in Node, and it probably has something to do with this.

In a separate project, I am using SWC to compile, so it works fine. Do I just need to also compile my Node project?

Thanks.

**Comment by fishcharlie** [comment]
@kevinswarner Sadly, I haven't worked much with Node modules. I'd be curious to know what happens if you just try to `console.log(dynamoose);`. Maybe you could also try `import dynamoose from 'dynamoose';`?

I'd just follow standard debugging practices here.

If you figure out what is wrong, I'd really appreciate if you could make a PR to update the Dynamoose documentation to help other users 😃

**Reply by kevinswarner** [reply]
I could not figure out a direct solution, so I moved to an approach that uses Babel to transpile my code to CommonJS, which seems to work. Thanks!
